modules/series-service-impl/src/main/java/org/opencastproject/series/impl/persistence/SeriesServiceDatabaseImpl.java

```diff
@@ -383,6 +403,60 @@ public DublinCoreCatalog getSeries(String seriesId) throws NotFoundException, Se
     }
   }
 
+  @Override
+  public List<Series> getAllForAdministrativeRead(Date from, Optional<Date> to, int limit)
+          throws SeriesServiceDatabaseException, UnauthorizedException {
+    // Validate parameters
+    if (limit <= 0) {
+      throw new IllegalArgumentException("limit has to be > 0");
+    }
+
+    // Make sure the user is actually an administrator of sorts
+    User user = securityService.getUser();
+    if (!user.hasRole(GLOBAL_ADMIN_ROLE) && !user.hasRole(user.getOrganization().getAdminRole())) {
+      throw new UnauthorizedException(user, getClass().getName() + ".getModifiedInRangeForAdministrativeRead");
+    }
+
+    // Load series from DB.
+    EntityManager em = emf.createEntityManager();
+    try {
+      TypedQuery<SeriesEntity> q;
+      if (to.isPresent()) {
+        if (from.after(to.get())) {
+          throw new IllegalArgumentException("`from` is after `to`");
+        }
+
+        q = em.createNamedQuery("Series.getAllModifiedInRange", SeriesEntity.class)
+            .setParameter("from", from)
+            .setParameter("to", to.get())
+            .setMaxResults(limit);
+      } else {
+        q = em.createNamedQuery("Series.getAllModifiedSince", SeriesEntity.class)
```

Comment:
Above, you check if the user is an admin or an organizational admin. This query returns all events from all organizations. This means, you allow admins from one organization to access events from another organization. I don't think that's intended, is it?

Reply:
This is now resolved in the query `getAllModifiedSince`